src/main/java/com/epam/preprod/sirenko/TwoListWrapper.java

```diff
@@ -0,0 +1,306 @@
+	 * from this list all of its elements that are not contained in the
+	 * specified collection.
+	 */
+	@Override
+	public boolean retainAll(Collection collection) {
+		if (!collection.containsAll(unmodifiedList)) {
+			throw new IllegalArgumentException("Can't retain collection of unmodified list");
+		} else {
+			modifiedList.retainAll(collection);
+		}
+		return true;
+	}
+	
+	/**
+	 * Removes all of the elements from modified list.
+	 */
+	@Override
+	public void clear() {
+		modifiedList.clear();
+	}
+	
+	/**
+	 * Returns the element at the specified position in this list.
+	 */
+	@Override
+	public E get(int index) {
+		if (index < unmodifiedList.size()) {
+			return unmodifiedList.get(index);
+		}
+		return modifiedList.get(index - unmodifiedList.size());
+	}
+	
+	/**
+	 * Replaces the element at the specified position in modified list with the
+	 * specified element (optional operation).
+	 */
+	@Override
+	public E set(int index, E element) {
+		if (index < unmodifiedList.size()) {
+			throw new IllegalArgumentException("Can't set element to unmodified list");
+		} else {
+			modifiedList.set(index - unmodifiedList.size(), element);
+		}
+		return modifiedList.get(index - unmodifiedList.size());
+	}
+	
+	/**
+	 * Inserts the specified element at the specified position in modified list
+	 * (optional operation).  Shifts the element currently at that position
+	 * (if any) and any subsequent elements to the right (adds one to their
+	 * indices).
+	 */
+	@Override
+	public void add(int index, E element) {
+		if (index < unmodifiedList.size()) {
+			throw new IllegalArgumentException("Can't add element to unmodified list");
+		} else {
+			modifiedList.add(index - unmodifiedList.size(), element);
+		}
+	}
+	
+	/**
+	 * Removes the element at the specified position in modified list.
+	 * Shifts any subsequent elements to the left (subtracts one
+	 * from their indices).  Returns the element that was removed from the
+	 * list.
+	 */
+	@Override
+	public E remove(int index) {
+		E toRemove;
+		if (index < unmodifiedList.size()) {
+			throw new IllegalArgumentException("Can't remove element to unmodified list");
+		} else {
+			toRemove = modifiedList.get(index - unmodifiedList.size());
+			modifiedList.remove(index - unmodifiedList.size());
+		}
+		return toRemove;
+	}
+	
+	/**
+	 * Returns the index of the first occurrence of the specified element
+	 * in this list, or -1 if this list does not contain the element.
+	 */
+	@Override
+	public int indexOf(Object object) {
+		if (unmodifiedList.contains(object)) {
+			return unmodifiedList.indexOf(object);
+		}
+		return modifiedList.indexOf(object) + unmodifiedList.size();
+	}
+	
+	/**
+	 * Returns the index of the last occurrence of the specified element
+	 * in this list, or -1 if this list does not contain the element.
+	 */
+	@Override
+	public int lastIndexOf(Object object) {
+		if (unmodifiedList.contains(object)) {
```

Comment:
unmodifiedList is before modified, but we looking for last index.So we need to check modifiedList first